### PR TITLE
[linstor] Introduce linstor-scheduler-admission

### PR DESCRIPTION
**Pull request by kvaps**
## Description

Introduce `linstor-scheduler-admission` for automatically setting `schedulerName` for Pods using linstor volumes.

## What is the expected result?

Now all pods using linstor volumes will automatically be configured to use linstor-scheduler

## Checklist
- [x] The code is covered by unit tests.
- [x] e2e tests passed.
- [x] Documentation updated according to the changes.
- [X] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: linstor
type: feature
summary: Introduce `linstor-scheduler-admission` for automatically setting `schedulerName` for Pods using linstor volumes.
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @kvaps. I've started the workflow for label 'e2e/run/static'...

File: modules/041-linstor/docs/USAGE_RU.md [hunk 1]
```diff
@@ -4,32 +4,9 @@ title: "Модуль linstor: примеры конфигурации"
...
-    ports:
-    - containerPort: 80
-  volumes:
-  - name: my-first-linstor-volume
-    persistentVolumeClaim:
-      claimName: "test-volume"
-```
+Планировщик `linstor` учитывает размещение данных в хранилище и старается размещать Pod в первую очередь на тех узлах, где данные доступны локально.  
+Любой Pod, использующий тома linstor, будет автоматически настроен на использование планировщика `linstor`.
+Any Pod using linstor volumes will be automatically configured to use the `linstor` scheduler.
```
Comment (z9r5) [inline modules/041-linstor/docs/USAGE_RU.md:9]:
```suggestion
Любой Pod, использующий тома linstor, будет автоматически настроен на использование планировщика `linstor`.
```

Comment (z9r5) [inline modules/041-linstor/docs/USAGE_RU.md:9]:
```suggestion
Любой Pod, использующий тома linstor, будет автоматически настроен на использование планировщика `linstor`.
```

**Review by z9r5 — DISMISSED** [review]

**Review by apolovov — CHANGES_REQUESTED** [review]
There are points.

File: modules/041-linstor/templates/linstor-scheduler-admission/webhook.yaml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+apiVersion: admissionregistration.k8s.io/v1
+kind: MutatingWebhookConfiguration
+metadata:
+  name: linstor-scheduler-admission
+  {{- include "helm_lib_module_labels" (list . (dict "app" "linstor-scheduler-admission")) | nindent 2 }}
+webhooks:
+- name: scheduler-admission.linstor.deckhouse.io
+  namespaceSelector:
+    matchExpressions:
+    - key: kubernetes.io/metadata.name
```
[the remark this answers is not in the document]
Comment (kvaps) [reply modules/041-linstor/templates/linstor-scheduler-admission/webhook.yaml:13]: internaly discussed to keep system namespaces excluded

File: tools/linstor_scheduler_webhook/main.go [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+	"text/template"
+)
+
+// Specify all d8-namespaces that might be controlled by linstor-scheduler
+var excludeNamespaces = []string{
```
[the remark this answers is not in the document]
Comment (kvaps) [reply tools/linstor_scheduler_webhook/main.go:14]: internaly discussed to keep system namespaces excluded

File: modules/041-linstor/templates/linstor-scheduler-admission/service.yaml [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+metadata:
+  name: linstor-scheduler-admission
+  namespace: d8-{{ .Chart.Name }}
+  {{- include "helm_lib_module_labels" (list . (dict "app" "linstor-scheduler-admission")) | nindent 2 }}
+spec:
+  selector:
+    app: linstor-scheduler-admission
+  ports:
+    - protocol: TCP
+      port: 8080
```
[the remark this answers is not in the document]
Comment (kvaps) [reply modules/041-linstor/templates/linstor-scheduler-admission/service.yaml:12]: done

File: tools/linstor_scheduler_webhook.go [hunk 5]
```diff
@@ -0,0 +1,19 @@
+/*
+Copyright 2022 Flant JSC
```
[the remark this answers is not in the document]
Comment (kvaps) [reply tools/linstor_scheduler_webhook.go:2]: fixed

File: modules/041-linstor/pod.yaml [hunk 6]
```diff
@@ -0,0 +1,29 @@
+---
```
[the remark this answers is not in the document]
Comment (kvaps) [reply modules/041-linstor/pod.yaml:1]: fixed

File: modules/041-linstor/hooks/generate_webhook_certs.go [hunk 7]
```diff
@@ -0,0 +1,119 @@
+/*
```
[the remark this answers is not in the document]
Comment (kvaps) [reply modules/041-linstor/hooks/generate_webhook_certs.go:1]: didn't know that, thanks,  fixed

**Review by z9r5 — APPROVED** [review]

**Review by apolovov — APPROVED** [review]

**Review by z9r5 — APPROVED** [review]

**Review by yalosev — APPROVED** [review]

**Pull request merged**